File: codeforces/1851A.cpp
// Escalator Conversations
#include <iostream>
#include <algorithm>
#include <cmath>

using namespace std;

int main()
{
    int numCases;
    cin >> numCases;

    for (int i = 0; i < numCases; i++)
    {
        int count = 0;
        int numPeople, numSteps, stepSize, h;
        cin >> numPeople >> numSteps >> stepSize >> h;

        int maxDiff = stepSize * (numSteps - 1);
        for (int j = 0; j < numPeople; j++)
        {
            int height;
            cin >> height;

            int diff = abs(height - h);
            if (diff <= maxDiff && height != h && diff % stepSize == 0)
                count++;
        }

        cout << count << endl;
    }

    return 0;
}

File: codeforces/1851A.py
num_cases = int(input())

for _ in range(num_cases):
    count = 0

    num_people, num_steps, step_size, h = map(int, input().split())
    heights = list(map(int, input().split()))

    max_diff = step_size * (num_steps - 1)
    for height in heights:
        diff = abs(height - h)
        if diff > max_diff:
            pass
        elif height != h and diff % step_size == 0:
            count += 1

    print(count)
